src/mpid/ch4/shm/ipc/src/ipc_p2p.h

```diff
@@ -51,17 +50,14 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPC_send_contig_lmt(const void *buf, MPI_Aint
     MPIDIG_REQUEST(sreq, count) = count;
     MPIDIG_REQUEST(sreq, context_id) = comm->context_id + context_offset;
 
-    MPIDI_Datatype_check_contig_size_lb(datatype, count, is_contig, data_sz, true_lb);
+    MPIDI_Datatype_check_contig_size(datatype, count, is_contig, data_sz);
     MPIR_Assert(is_contig && data_sz > 0);
 
```

Comment:
How about we store `is_contig` and `data_sz` in `mem_attr`? In the current code, it looks like we have to calculate such datatype info at least twice even in the fast contig path.

Reply:
Adding these to the `mem_attr` is overloading that with mixed functionality that makes it lesser clean. Actually we don't need `is_contig` at all here because the function is only supposed to be called for contig buffers. We can add a parameter to the function with `data_sz` as well to get rid of the extra datatype query.